polkadot/runtime/parachains/src/paras/mod.rs

```diff
@@ -1491,6 +1624,27 @@ impl<T: Config> Pallet<T> {
 		T::DbWeight::get().reads_writes(1 + pruning_tasks_done, 2 * pruning_tasks_done)
 	}
 
+	/// This function removes authorizations that have expired,
+	/// meaning their `expire_at` block is less than or equal to the current block (`now`).
+	fn prune_expired_authorizations(now: BlockNumberFor<T>) -> Weight {
+		let mut weight = T::DbWeight::get().reads(1);
+		let to_remove = AuthorizedCodeHash::<T>::iter().filter_map(
+			|(para, AuthorizedCodeHashAndExpiry { expire_at, .. })| {
+				if expire_at <= now {
+					Some(para)
+				} else {
+					None
+				}
+			},
+		);
+		for para in to_remove {
+			AuthorizedCodeHash::<T>::remove(&para);
+			weight.saturating_accrue(T::DbWeight::get().writes(1));
```

Comment:
I believe in reality it's just one write to db, I think all these removes get batched in a single DB write